pkg/corerp/backend/controller/createorupdateresource.go

```diff
@@ -125,6 +127,8 @@ func (c *CreateOrUpdateResource) Run(ctx context.Context, request *ctrl.Request)
 		}
 	}
 
+	// We persist the resource to the store after the deployment is done successfully as well as unsuccessfully so that Radius can still manage the resource.
+	// For example, even though a deployment failed, we still want to be able to delete the resource through rad app delete.
```

Comment:
We should test and see if this could have negative effects on our flow right now.